gsoc/2017.md

```diff
@@ -0,0 +1,150 @@
+    long, but in reality it is very easy to run out of time).
+*   The [official SoC timetable](https://summerofcode.withgoogle.com/how-it-works/#timeline)
+    mentions May 30 as the official start of coding, but if you
+    have time you are encouraged to research your proposals even
+    before that (and definitely learn the basics of Scala, if you
+    haven't done that already).
+
+### Student Application Guidelines
+
+*   Student proposals should be very specific. We want to see evidence
+    that you can succeed in the project. Applications with one-liners
+    and general descriptions definitely won't make the cut.
+*   Because of the nature of our projects students must have at some
+    knowledge of the Scala language. Applicants with Scala programming
+    experience will be preferred. Alternatively, experience with
+    functional programming could suffice, but in your application we
+    want to see evidence that you can quickly be productive in Scala.
+*   You can think of Google Summer of Code as a kind of independent
+    internship. Therefore, we expect you to work full-time during the
+    duration. Applicants with other time commitments are unlikely to
+    be selected. From our previous experience we know that students'
+    finishing their studies (either Bachelor, Master of PhD) are
+    likely to be overwhelmed by their final work, so please don't be
+    too optimistic and carefully plan your time for the project.
+*   If you are unsure whether your proposal is suitable, feel free to
+    discuss it on our
+    ["contributors"](http://contributors.scala-lang.org/)
+    mailing list. We have many community members on our mailing list
+    who will quickly answer any of your questions regarding the
+    project. Mentors are also constantly monitoring the mailing
+    list. Don't be afraid of asking questions, we enjoy solving
+    puzzles like that!
+
+### General Proposal Requirements
+
+The proposal will be submitted via the standard web-interface at
+[https://summerofcode.withgoogle.com/](https://summerofcode.withgoogle.com/),
+therefore plain text is the best way to go. We expect your application
+to be in the range of 700-1500 words. Anything less than that will
+probably not contain enough information for us to determine whether
+you are the right person for the job.
+
+Your proposal should contain at least the following information, but
+feel free to include anything that you think is relevant:
+
+*   Please include your name (weird as it may be, people do forget
+    about it)
+*   Title of your proposal
+*   Abstract of your proposal
+*   Detailed description of your idea including explanation on why is
+    it innovative (maybe you already have some prototype?), what
+    contribution do you expect to make to the Scala community and why
+    do you think your project is needed, a rough plan of your
+    development and possible architecture sketches.
```

Comment:
It feels weird to say "development" in this context. Maybe that's just me, @heathermiller ?